robosistance/src/endpoints/robot.rs

```diff
@@ -26,19 +31,43 @@ impl fmt::Display for Action {
     }
 }
 
-
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct Command {
-    action: Action,
-    argument: u64, // FIXME: This should be some sort of generic byte string
+    pub action: Action,
+    pub argument: u64, // FIXME: This should be some sort of generic byte string
 }
 
-
 #[get("/command")]
-fn serve_command() { // Stream of commands or something that contains commands
-    // Provide a buffer of commands for the robot to execute
-}
+pub fn establish_connection(
+    active_queues: &State<RwLock<HashMap<Uuid, Sender<Command>>>>,
+    mut end: Shutdown,
+) -> EventStream![] {
+    //! Establish connection with the robot and send any commands that are added to a buffer of commands to the robot.
 
+    // Initialise a send queue and subscribe to that queue
+    let command_queue: Sender<Command> = channel::<Command>(1024).0;
+    let mut receiver: Receiver<Command> = command_queue.subscribe();
+
+    // Lock the active queues hashmap before inserting a sender queue of commands
+    let mut locked_hashmap = active_queues.write().unwrap();
```

Comment:
This is okay for now, but should ideally be handled by something that isn't unwrap.